docs/server-learner.rst

```diff
@@ -140,6 +142,25 @@ etcd server must not transfer leadership to learner, since it may still lag behi
 
 Internally in Raft, second ``MemberAdd`` call to learner node promotes it to a voting member. Leader maintains the progress of each follower and learner. If learner has not completed its snapshot message, reject promote request. Only accept promote request if and only if: The learner node is in a healthy state. The learner is in sync with leader or the delta is within the threshold (e.g. the number of entries to replicate to learner is less than 1/10 of snapshot count, which means it is less likely that even after promotion leader would not need send snapshot to the learner). All these logic are hard-coded in ``etcdserver`` package and not configurable.
 
+*Expose "Auto-promoting" node type to "MemberAdd" API.*
+
+etcd client adds an “AutoPromote” flag to ``MemberAdd`` API for auto-promoting learner nodes. This flag is only effective when used in conjunction with the “IsLearner” flag, and is ignored otherwise. An etcd server handler applies membership change entry with ``pb.ConfChangeAddAutoPromotingNode`` type. Once the command has been applied, a server joins the cluster with ``etcd --initial-cluster-state=existing`` flag.
```

Comment:
Is "AutoPromote" flag only needed in short term and will eventually be deprecated?

Reply:
Hey @jingyih, I think you could definitely make a case for that. 

On the other hand, I think you could also make a case for removing the `—learner` (and `IsLearner`) flag, because in v3.5 the default and only behavior of of newly added members is to be a learner (until manually or automatically promoted). It might be more expressive to keep `—auto-promote` and allow operators to specify `—auto-promote=false` to add a standby learner.